hs_file_types/tests/test_geofeature_metadata.py

```diff
@@ -0,0 +1,506 @@
+        self.states_zip_invalid_file_name = 'states_invalid.zip'
+        base_file_path = 'hs_file_types/tests/data/{}'
+        self.states_required_zip_file = base_file_path.format(self.states_required_zip_file_name)
+        self.states_prj_file = base_file_path.format(self.states_prj_file_name)
+        self.states_xml_file = base_file_path.format(self.states_xml_file_name)
+        self.states_shp_file = base_file_path.format(self.states_shp_file_name)
+        self.states_dbf_file = base_file_path.format(self.states_dbf_file_name)
+        self.states_shx_file = base_file_path.format(self.states_shx_file_name)
+        self.osm_all_files_zip_file = base_file_path.format(self.osm_all_files_zip_file_name)
+        self.states_zip_invalid_file = base_file_path.format(self.states_zip_invalid_file_name)
+
+        self.allowance = 0.00001
+
+    def tearDown(self):
+        super(GeoFeatureFileTypeMetaDataTest, self).tearDown()
+        if os.path.exists(self.temp_dir):
+            shutil.rmtree(self.temp_dir)
+
+    def test_zip_set_file_type_to_geo_feature_required(self):
+        # here we are using a zip file that has only the 3 required files for setting it
+        # to Geo Feature file type which includes metadata extraction
+
+        self._create_composite_resource(self.states_required_zip_file)
+
+        self.assertEqual(self.composite_resource.files.all().count(), 1)
+        res_file = self.composite_resource.files.first()
+        expected_folder_name = res_file.file_name[:-4]
+        # check that the resource file is associated with GenericLogicalFile
+        self.assertEqual(res_file.has_logical_file, True)
+        self.assertEqual(res_file.logical_file_type_name, "GenericLogicalFile")
+        # check that there is one GenericLogicalFile object
+        self.assertEqual(GenericLogicalFile.objects.count(), 1)
+        res_file = self.composite_resource.files.first()
+        self.assertEqual(res_file.extension, '.zip')
+        logical_file = res_file.logical_file
+        self.assertTrue(isinstance(logical_file.metadata, GenericFileMetaData))
+        self.assertTrue(isinstance(logical_file, GenericLogicalFile))
+        # TODO: not sure why there would be file level keywords at this point - commenting to
+        # avoid test failure
+        # self.assertEqual(logical_file.metadata.keywords, [])
+
+        # set the zip file to GeoFeatureFile type
+        GeoFeatureLogicalFile.set_file_type(self.composite_resource, res_file.id, self.user)
+
+        # test file type and file type metadata
+        assert_geofeature_file_type_metadata(self, expected_folder_name)
+
+        # there should not be any file level keywords
+        # TODO: not sure why there would be file level keywords - commented out as the test is
+        # failing
+        # res_file = self.composite_resource.files.first()
+        # logical_file = res_file.logical_file
+        # self.assertEqual(logical_file.metadata.keywords, [])
```

Comment:
@pkdash If this is still an issue, it'd be good to get to the bottom of it, i.e., why there are file level keywords when there shouldn't be?